sdk/cosmosdb/cosmos/test/internal/unit/query/continuationToken/orderByQueryContinuationToken.spec.ts

```diff
@@ -0,0 +1,391 @@
+          undefined as any,
+          mockRid,
+          mockSkipCount,
+        );
+      }, "orderByItems must contain at least one element");
+    });
+
+    it("should throw error when both arrays are empty", () => {
+      assert.throws(() => {
+        createOrderByQueryContinuationToken([], [], mockRid, mockSkipCount);
+      }, "rangeMappings must contain at least one element");
+    });
+
+    it("should create a token with zero skip count", () => {
+      const token = createOrderByQueryContinuationToken(
+        mockRangeMappings,
+        mockOrderByItems,
+        mockRid,
+        0,
+      );
+
+      assert.deepEqual(token.rangeMappings, mockRangeMappings);
+      assert.deepEqual(token.orderByItems, mockOrderByItems);
+      assert.equal(token.rid, mockRid);
+      assert.equal(token.skipCount, 0);
+    });
+
+    it("should create a token with complex order by items", () => {
+      const complexOrderByItems = [
+        { value: null, type: "null" },
+        { value: true, type: "boolean" },
+        { value: [1, 2, 3], type: "array" },
+        { value: { nested: "object" }, type: "object" },
+      ];
+
+      const token = createOrderByQueryContinuationToken(
+        mockRangeMappings,
+        complexOrderByItems,
+        mockRid,
+        mockSkipCount,
+      );
+
+      assert.deepEqual(token.orderByItems, complexOrderByItems);
+    });
+  });
+
+  describe("serializeOrderByQueryContinuationToken", () => {
+    it("should serialize token to JSON string", () => {
+      const token = createOrderByQueryContinuationToken(
+        mockRangeMappings,
+        mockOrderByItems,
+        mockRid,
+        mockSkipCount,
+        mockDocumentRid,
+        5,
+        20,
+        "hash123",
+      );
+
+      const serialized = serializeOrderByQueryContinuationToken(token);
+      const parsed = JSON.parse(serialized);
+
+      assert.equal(parsed.rid, mockRid);
+      assert.equal(parsed.skipCount, mockSkipCount);
+      assert.equal(parsed.documentRid, mockDocumentRid);
+      assert.equal(parsed.offset, 5);
+      assert.equal(parsed.limit, 20);
+      assert.equal(parsed.hashedLastResult, "hash123");
+      assert.equal(parsed.rangeMappings.length, 2);
+      assert.equal(parsed.orderByItems.length, 2);
+    });
+
+    it("should serialize token without optional parameters", () => {
+      const token = createOrderByQueryContinuationToken(
+        mockRangeMappings,
+        mockOrderByItems,
+        mockRid,
+        mockSkipCount,
+      );
+
+      const serialized = serializeOrderByQueryContinuationToken(token);
+      const parsed = JSON.parse(serialized);
+
+      assert.equal(parsed.rid, mockRid);
+      assert.equal(parsed.skipCount, mockSkipCount);
+      assert.isUndefined(parsed.documentRid);
+      assert.isUndefined(parsed.offset);
+      assert.isUndefined(parsed.limit);
+      assert.isUndefined(parsed.hashedLastResult);
+    });
+
+    it("should serialize token with complex order by items", () => {
```

Comment:
do we even support cross type/ object type order by queries?

Reply:
not now but we might do it in future, and this test is for checking serialization.